src/components/field/index.js

```diff
@@ -324,74 +345,78 @@ export default class TextField extends PureComponent {
   }
 
   setValue(text) {
+
     this.setState({ text })
   }
 
   isFocused() {
+
     let input = this.correctInputRef();
 
     return input.isFocused()
   }
 
   isRestricted() {
+
     let { characterRestriction: limit } = this.props
     let { length: count } = this.value()
 
     return limit < count
   }
 
   isErrored() {
+
     return errorStateFromProps(this.props)
   }
 
   isDefaultVisible() {
+
     let { text, receivedFocus } = this.state
     let { defaultValue } = this.props
-
     return !receivedFocus && text == null && defaultValue != null
   }
 
   isPlaceholderVisible() {
+
     let { placeholder } = this.props
 
     return placeholder && !this.focused && !this.value()
   }
 
   isLabelActive() {
+
     return this.labelState() === 1
   }
 
   onFocus(event) {
+
     let { onFocus, clearTextOnFocus } = this.props
     let { receivedFocus } = this.state
 
     if (typeof onFocus === 'function') {
       onFocus(event)
     }
 
-    if (clearTextOnFocus) {
-      this.clear()
-    }
 
     this.focused = true
 
     this.startFocusAnimation()
     this.startLabelAnimation()
 
     if (!receivedFocus) {
-      this.setState({ receivedFocus: true, text: this.value() })
+      this.setState({ receivedFocus: true, text: this.value(), extractedText: this.value()})
```

Comment:
I'm hoping to get some input from @amy-at-braid on this one. I feel like there should be a better way to get extracted text, but I'm also unwilling to expose it from the underlying mask input component. Any ideas?

Reply:
I'm a little confused here, shouldn't `extractedText` and `text` be different? I assume `text` is formatted whereas `extractedText` isn't, right? It's been a while since I've looked at this though.

Reply:
text is formatted but extractedText isn't, so text would contain a "$" but extractedText would not

Reply:
I'm a little confused then - why are they both equal to `this.value()`?

Reply:
They should be, but I don't have any good ideas on how to apply the mask algorithm to the text and obtain extracted text properly. Do you have any ideas, @amy-at-braid